blsSignatures/blsSignatures.go

```diff
@@ -118,31 +123,36 @@ func verifySignature2(sig Signature, message []byte, publicKey PublicKey, keyVal
 		return false, err
 	}
 
-	engine := bls12381.NewEngine()
-	engine.Reset()
-	engine.AddPair(pointOnCurve, publicKey.key)
-	leftSide := engine.Result()
-	engine.AddPair(sig, engine.G2.One())
-	rightSide := engine.Result()
-	return leftSide.Equal(rightSide), nil
+	leftSide, err := bls12381.Pair([]bls12381.G1Affine{*pointOnCurve}, []bls12381.G2Affine{*publicKey.key})
+	if err != nil {
+		return false, err
+	}
+	_, _, _, g2 := bls12381.Generators()
+	rightSide, err := bls12381.Pair([]bls12381.G1Affine{*sig}, []bls12381.G2Affine{g2})
+	if err != nil {
+		return false, err
+	}
+	return leftSide.Equal(&rightSide), nil
 }
 
 func AggregatePublicKeys(pubKeys []PublicKey) PublicKey {
-	g2 := bls12381.NewG2()
-	ret := g2.Zero()
+	g2 := new(bls12381.G2Affine)
+	g2.X.SetZero()
+	g2.Y.SetZero()
 	for _, pk := range pubKeys {
-		g2.Add(ret, ret, pk.key)
+		g2.Add(g2, pk.key)
 	}
-	return NewTrustedPublicKey(ret)
+	return NewTrustedPublicKey(g2)
 }
 
 func AggregateSignatures(sigs []Signature) Signature {
-	g1 := bls12381.NewG1()
-	ret := g1.Zero()
+	g1 := new(bls12381.G1Affine)
+	g1.X.SetZero()
+	g1.Y.SetZero()
```

Comment:
Same thing here, I don't think we need SetZero

Reply:
Done.